src/util/exportUtils.js

```diff
@@ -2,84 +2,37 @@ const { BadRequestError } = require('./errorUtils');
 const logger = require('../server/logger');
 
 /**
- * Uses request body parameter to search for the export server URL. Validates that
- * only one URL is present.
+ * Uses request body parameter to get all of the ndjson URLs
  * @param {Object} parameters - request body parameter
- * @returns export server URL string
+ * @returns array of ndjson fileUrls
  */
-const retrieveExportUrl = parameters => {
-  logger.debug(`Retrieving export URL from parameters: ${JSON.stringify(parameters)}`);
-  const exportUrlArray = parameters.filter(param => param.name === 'exportUrl');
-  checkExportUrlArray(exportUrlArray);
-  let exportUrl = exportUrlArray[0].valueUrl;
-
-  // Retrieve comma-delimited list of type filters from parameters
-  const typesString = parameters
-    .filter(param => param.name === '_type')
-    .map(function (type) {
-      logger.debug(`Adding type ${type} to exportUrl type parameter`);
-      return type.valueString;
-    })
-    .toString();
-
-  const typeFilterString = parameters
-    .filter(param => param.name === '_typeFilter')
-    .map(function (typeFilter) {
-      logger.debug(`Adding typeFilter ${typeFilter} to exportUrl typeFilter parameter`);
-      return typeFilter.valueString;
-    })
-    .toString();
-
-  if (typesString) {
-    if (exportUrl.includes(`_type=`)) {
-      console.warn('_type parameter already supplied in exportUrl. Omitting entries from parameter array');
-    } else {
-      // add types from parameters to exportUrl
-      exportUrl += `${exportUrl.includes('_typeFilter=') ? '&' : '?'}_type=${typesString}`;
-    }
-  }
-
-  if (typeFilterString) {
-    if (exportUrl.includes(`_typeFilter=`)) {
-      console.warn('_typeFilter parameter already supplied in exportUrl. Omitting entries from parameter array');
-    } else {
-      // add type filters from parameters to exportUrl
-      exportUrl += `${exportUrl.includes('_type=') ? '&' : '?'}_typeFilter=${typeFilterString}`;
-    }
-  }
-  return exportUrl;
+const retrieveInputUrls = parameters => {
+  logger.debug(`Retrieving all input URLs from parameters: ${JSON.stringify(parameters)}`);
+  const inputParamArray = parameters.filter(param => param.name === 'input');
```

Comment:
I noticed that if I don't include the `Content-Type: application/fhir+json` header in the request, the code ends up here with `parameters` being undefined as if something didn't allow the body to pass when the content was wrong instead of responding that the content type was unexpected.

Reply:
I see! Maybe we can discuss this- I am not sure what level this error should be thrown? I am not sure if the spec is clear whether this should be an error thrown during the kick-off request or if it should be a transient error that occurs during the $import that we then see in the bulkstatus response (the latter is what we currently do).

Reply:
It looks like we already have a utility for this in [baseUtils.js](https://github.com/projecttacoma/deqm-test-server/blob/dc89180bf4b355e97db38cfa84471dd16db82e93/src/util/baseUtils.js#L32). Just have to use it in the `import.service.js` where the request is initially being handled with a simple `checkContentTypeHeader(req.headers);`